samples/bluetooth/fast_pair/locator_tag/src/ui/platform_dk.c

```diff
@@ -134,145 +149,100 @@ static void bootup_btn_handle(void)
+		(void) k_work_reschedule_for_queue(&led_workq, dwork, K_MSEC(sleep_time_ms));
+	} else {
+		run_led_on = false;
 	}
-}
 
-K_THREAD_DEFINE(run_led_thread_id, RUN_LED_THREAD_STACK_SIZE, run_led_thread_process,
-		NULL, NULL, NULL, RUN_LED_THREAD_PRIORITY, 0, 0);
+	dk_set_led(APP_RUN_STATUS_LED, run_led_on);
+}
 
-static void mode_led_thread_process(void)
+static void mode_led_work_handle(struct k_work *item)
 {
+	struct k_work_delayable *dwork = k_work_delayable_from_work(item);
 	static bool mode_led_on;
 
-	while (1) {
-		uint32_t sleep_time_ms = MODE_LED_THREAD_BACKOFF_TIMEOUT_MS;
-		enum app_state_mode mode = atomic_get(&app_state_mode);
+	bool id_mode = atomic_test_bit(ui_state_status, APP_UI_STATE_ID_MODE);
+	bool recovery_mode = atomic_test_bit(ui_state_status, APP_UI_STATE_RECOVERY_MODE);
 
-		switch (mode) {
-		case APP_STATE_MODE_NONE:
-			if (mode_led_on) {
-				mode_led_on = false;
-				dk_set_led(APP_MODE_LED, mode_led_on);
-			}
-			break;
-
-		case APP_STATE_MODE_RECOVERY:
-			mode_led_on = !mode_led_on;
-			dk_set_led(APP_MODE_LED, mode_led_on);
-			sleep_time_ms = MODE_LED_RECOVERY_BLINK_INTERVAL_MS;
-			break;
-
-		case APP_STATE_MODE_ID:
-			mode_led_on = !mode_led_on;
-			dk_set_led(APP_MODE_LED, mode_led_on);
-			sleep_time_ms = MODE_LED_ID_BLINK_INTERVAL_MS;
-			break;
-
-		case APP_STATE_MODE_ALL:
-			if (!mode_led_on) {
-				mode_led_on = true;
-				dk_set_led(APP_MODE_LED, mode_led_on);
-			}
-			break;
-
-		default:
-			__ASSERT_NO_MSG(false);
-			break;
-		}
+	if (recovery_mode != id_mode) {
+		uint32_t sleep_time_ms = recovery_mode ?
+					 MODE_LED_RECOVERY_BLINK_INTERVAL_MS :
+					 MODE_LED_ID_BLINK_INTERVAL_MS;
 
-		k_sleep(K_MSEC(sleep_time_ms));
+		mode_led_on = !mode_led_on;
+		(void) k_work_reschedule_for_queue(&led_workq, dwork, K_MSEC(sleep_time_ms));
+	} else {
+		mode_led_on = recovery_mode && id_mode;
 	}
-}
 
-K_THREAD_DEFINE(mode_led_thread_id, MODE_LED_THREAD_STACK_SIZE, mode_led_thread_process,
-		NULL, NULL, NULL, MODE_LED_THREAD_PRIORITY, 0, 0);
+	dk_set_led(APP_MODE_LED, mode_led_on);
+}
 
-static void state_app_running_indicate(bool active)
+static void provisioning_led_work_handle(struct k_work *item)
 {
-	atomic_set(&run_led_thread_state_app_running, active);
+	struct k_work_delayable *dwork = k_work_delayable_from_work(item);
+	static bool provisioning_led_on;
 
-	if (!active) {
-		dk_set_led(APP_RUN_STATUS_LED, false);
-	}
-}
+	bool provisioned = atomic_test_bit(ui_state_status, APP_UI_STATE_PROVISIONED);
+	bool fp_adv = atomic_test_bit(ui_state_status, APP_UI_STATE_FP_ADV);
 
-static void state_mode_indicate(enum app_ui_state state, bool active)
-{
-	static bool recovery_mode;
-	static bool id_mode;
-	enum app_state_mode mode;
-
-	switch (state) {
-	case APP_UI_STATE_RECOVERY_MODE:
-		recovery_mode = active;
-		break;
-	case APP_UI_STATE_ID_MODE:
-		id_mode = active;
-		break;
-	default:
-		__ASSERT_NO_MSG(false);
-		break;
-	}
+	if (fp_adv) {
+		uint32_t sleep_time_ms = provisioned ?
```

Comment:
`blink_interval_ms` to align the variable name with definitions (also in three other functions)